1436790/1438462: Part 2: 10 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
For:
* [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
* [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037).

DennisLee-DennisLee committed:
10 files with MSDN link fixes.

DennisLee-DennisLee commented:
#sign-off

@mairaw, should I at-mention you in all of my PRs, or do you just want to leave the approvals/rejections to whoever gets autonotified for review?

rpetrusha reviewed (changes requested):
Thanks for replacing these MSDN links, @DennisLee-DennisLee. I've left a suggested change, and also thought that two of the links should be eliminated entirely.

mairaw commented:
You don't need to tag me @DennisLee-DennisLee for this work. Whoever sees it first will review.

mairaw commented on docs/csharp/programming-guide/concepts/async/index.md:
```diff
@@ -238,7 +238,7 @@ Asynchronous APIs in Windows Runtime programming have one of the following retur
 |[Fine-Tuning Your Async Application (C#)](../../../../csharp/programming-guide/concepts/async/fine-tuning-your-async-application.md)|Shows how to add the following functionality to your async solution:<br /><br /> -   [Cancel an Async Task or a List of Tasks (C#)](../../../../csharp/programming-guide/concepts/async/cancel-an-async-task-or-a-list-of-tasks.md)<br />-   [Cancel Async Tasks after a  ...
 |[Handling Reentrancy in Async Apps (C#)](../../../../csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md)|Shows how to handle cases in which an active asynchronous operation is restarted while it’s running.||  
-|[WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140(v=vs.120).aspx)|Shows how to bridge between Task types in the .NET Framework and IAsyncOperations in the [!INCLUDE[wrt](~/includes/wrt-md.md)] so that you can use <xref:System.Threading.Tasks.Task.WhenAny%2A> with a [!INCLUDE[wrt](~/includes/wrt-md.md)] method.|[Async Sample: Bridgi ...
+|[WhenAny: Bridging between the .NET Framework and the Windows Runtime](/previous-versions/visualstudio/visual-studio-2013/jj635140(v=vs.120))|Shows how to bridge between Task types in the .NET Framework and IAsyncOperations in the [!INCLUDE[wrt](~/includes/wrt-md.md)] so that you can use <xref:System.Threading.Tasks.Task.WhenAny%2A> with a [!INCLUDE[wrt](~/includes/wrt-md.md)] method.|[Async Sam ...
```
I've added this case to the task last night, but let's keep the domain on previous-versions links. We only remove it for the content sets that we know have offline books.

```suggestion
|[WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/jj635140(v=vs.120))|Shows how to bridge between Task types in the .NET Framework and IAsyncOperations in the [!INCLUDE[wrt](~/includes/wrt-md.md)] so that you can use <xref:System.Threading.Tasks.Task.WhenAny%2A> with a [!INCLUDE[wrt](~/includes/wrt-md.md)] method.|[Async Sample: Bridging between .NET and Windows Runtime (AsTask and WhenAny)](https://code.msdn.microsoft.com/Async-Sample-Bridging-d6a2f739)|  
```

mairaw commented on docs/csharp/programming-guide/concepts/collections.md:
```diff
@@ -577,7 +577,7 @@ private static IEnumerable<int> EvenSequence(
 - [Parallel LINQ (PLINQ)](../../../standard/parallel-programming/parallel-linq-plinq.md)
 - [Collections and Data Structures](../../../standard/collections/index.md)
-- [Creating and Manipulating Collections](https://msdn.microsoft.com/library/2065398e-eb1a-4821-9188-75f16e42e069)
+- [Creating and Manipulating Collections](/previous-versions/dotnet/netframework-4.0/14ek9axh(v=vs.100))
```
```suggestion
- [Creating and Manipulating Collections](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/14ek9axh(v=vs.100))
```

mairaw commented on docs/csharp/programming-guide/delegates/index.md:
```diff
@@ -40,7 +40,7 @@ A [delegate](../../../csharp/language-reference/keywords/delegate.md) is a type
 -   [Using Delegates](../../../csharp/programming-guide/delegates/using-delegates.md)  
   
--   [When to Use Delegates Instead of Interfaces (C# Programming Guide)](https://msdn.microsoft.com/library/2e759bdf-7ca4-4005-8597-af92edf6d8f0)  
+-   [When to Use Delegates Instead of Interfaces (C# Programming Guide)](/previous-versions/visualstudio/visual-studio-2010/ms173173(v=vs.100))  
```
```suggestion
-   [When to Use Delegates Instead of Interfaces (C# Programming Guide)](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2010/ms173173(v=vs.100))  
```

mairaw commented on docs/csharp/programming-guide/types/casting-and-type-conversions.md:
```diff
@@ -91,6 +91,6 @@ Giraffe g2 = (Giraffe) a;
 - [Conversion Operators](../../../csharp/programming-guide/statements-expressions-operators/conversion-operators.md)
-- [Generalized Type Conversion](https://msdn.microsoft.com/library/49253ae6-7657-4810-82ab-1176a6feeada)
-- [Exported Type Conversion](https://msdn.microsoft.com/library/1dfe55f4-07a2-4b61-aabf-a8cf65783a6b)
+- [Generalized Type Conversion](/previous-versions/visualstudio/visual-studio-2013/yy580hbd(v=vs.120))
```
```suggestion
- [Generalized Type Conversion](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/yy580hbd(v=vs.120))
```

mairaw commented on docs/framework/64-bit-apps.md:
```diff
@@ -32,7 +32,7 @@ When you compile an application, you can specify that it should run on a Windows
 -   Code that casts `IntPtr` to `Int32`.  
   
- For more information about how to port a 32-bit application to run on the 64-bit CLR, see [Migrating 32-bit Managed Code to 64-bit](https://msdn.microsoft.com/library/ms973190.aspx).  
+ For more information about how to port a 32-bit application to run on the 64-bit CLR, see [Migrating 32-bit Managed Code to 64-bit](/previous-versions/dotnet/articles/ms973190(v=msdn.10)).  
```
```suggestion
 For more information about how to port a 32-bit application to run on the 64-bit CLR, see [Migrating 32-bit Managed Code to 64-bit](https://docs.microsoft.com/previous-versions/dotnet/articles/ms973190(v=msdn.10)).  
```

mairaw commented on docs/framework/app-domains/how-to-load-assemblies-into-an-application-domain.md:
```diff
@@ -31,7 +31,7 @@ There are several ways to load an assembly into an application domain. The recom
 >  Starting with the .NET Framework version 2.0, the runtime will not load an assembly that was compiled with a version of the .NET Framework that has a higher version number than the currently loaded runtime. This applies to the combination of the major and minor components of the version number.  
   
- You can specify the way the just-in-time (JIT) compiled code from loaded assemblies is shared between application domains. For more information, see [Application Domains and Assemblies](https://msdn.microsoft.com/library/433b04ae-4ba8-4849-9dbd-79194f240346).  
+ You can specify the way the just-in-time (JIT) compiled code from loaded assemblies is shared between application domains. For more information, see [Application Domains and Assemblies](/previous-versions/dotnet/netframework-4.0/43wc4hhs(v=vs.100)).  
```
```suggestion
 You can specify the way the just-in-time (JIT) compiled code from loaded assemblies is shared between application domains. For more information, see [Application domains and assemblies](application-domains.md#application-domains-and-assemblies).  
```

mairaw commented on docs/framework/app-domains/how-to-load-assemblies-into-an-application-domain.md:
```diff
@@ -46,4 +46,4 @@ There are several ways to load an assembly into an application domain. The recom
 - [Using Application Domains](../../../docs/framework/app-domains/use.md)
 - [How to: Load Assemblies into the Reflection-Only Context](../../../docs/framework/reflection-and-codedom/how-to-load-assemblies-into-the-reflection-only-context.md)
-- [Application Domains and Assemblies](https://msdn.microsoft.com/library/433b04ae-4ba8-4849-9dbd-79194f240346)
+- [Application Domains and Assemblies](/previous-versions/dotnet/netframework-4.0/43wc4hhs(v=vs.100))
```
```suggestion
- [Application domains and assemblies](application-domains.md#application-domains-and-assemblies)
```

mairaw commented on docs/framework/app-domains/how-to-view-assembly-contents.md:
```diff
@@ -83,6 +83,6 @@ You can use the [Ildasm.exe (IL Disassembler)](../../../docs/framework/tools/ild
   
 ## See also
-- [Application Domains and Assemblies](https://msdn.microsoft.com/library/433b04ae-4ba8-4849-9dbd-79194f240346)
+- [Application Domains and Assemblies](/previous-versions/dotnet/netframework-4.0/43wc4hhs(v=vs.100))
```
```suggestion
- [Application domains and assemblies](application-domains.md#application-domains-and-assemblies)
```

mairaw reviewed (changes requested):
Looks good @DennisLee-DennisLee. Left some suggestions to be addressed in addition to @rpetrusha's comments. You can batch them via the Files tab.

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-002

DennisLee-DennisLee committed:
Apply suggestions from code review

Accepted most of the suggestions. Still need to review a couple of previous version links that show "dotnet" as the child repo. Also need to do some link deletions manually.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Merge branch 'v-dele-1438462-002' of https://github.com/DennisLee-DennisLee/docs into v-dele-1438462-002

mairaw commented:
There are still some suggestions pending @DennisLee-DennisLee. Can you please take a look?

mairaw committed:
Apply suggestions from code review

Accepting suggested change to prepend "https://docs.microsoft.com" to the two new "/previous-versions/dotnet/…" links.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Merge branch 'v-dele-1438462-002' of https://github.com/DennisLee-DennisLee/docs into v-dele-1438462-002

DennisLee-DennisLee committed:
Incorporating Ron's two suggested deletions.

DennisLee-DennisLee commented:
Sorry @mairaw, I wanted to review some of the comments separately instead of batching all of the suggestions at once. I hope I didn't wake you up. :)

mairaw commented:
Haha no worries and no, I wouldn't wake up for this 😄 but I work late more than what I'd like LOL

mairaw reviewed (commented):
One last change I noticed and we're good to merge. Thanks @DennisLee-DennisLee!

DennisLee-DennisLee committed:
Removing the "Creating and Manipulating Collections" link from 2 files.

DennisLee-DennisLee commented:
#sign-off

mairaw reviewed (approved):
Looks great @DennisLee-DennisLee! 🆒

rpetrusha commented:
Thanks, @DennisLee-DennisLee. We'll merge now.
